### DISCUSSION TITLE
FATAL   node    preparing POST initializer failed

### DISCUSSION THREAD

**Discussion by honglouzzc** [spacemeshos/go-spacemesh#5790] [Q&A] [answered]
when executing
./go-spacemesh --config config.mainnet.json --smeshing-start --smeshing-coinbase sm1xxxxxxx --smeshing-opts-numunits 16 --smeshing-opts-provider 4294967295 --smeshing-opts-datadir /data/ post_data --data-folder /data/node_data
  An error message appears as shown in the screenshot
![b36e8255e9f034ea3ba7a004844928e7](https://github.com/spacemeshos/go-spacemesh/assets/52514907/468a3226-2aec-4bcf-aa6c-d37d066e705f)

**Comment by poszu** [comment, marked as answer]
Hi @honglouzzc, the error means that the identity key file, located in data-dir/identities (probably in /mnt/data/spacemesh/identities/identity.key) doesn't match the public key (in /mnt/data/spacemesh/post_data). Did you change go-spacemesh versions 1.4 -> 1.3 -> 1.4?

**Reply by honglouzzc** [reply]
Thank you very much. It is caused by the inconsistency of node id. The key location was redefined in version 1.4.0.
![20240331-084138](https://github.com/spacemeshos/go-spacemesh/assets/52514907/f9c57326-5ea6-4a6e-90dc-2c47c8a1c18b)
